lib/eslint/eslint.js

```diff
@@ -314,7 +357,33 @@ async function loadFlatConfigFile(filePath) {
         delete require.cache[filePath];
     }
 
-    const config = (await import(fileURL)).default;
+    const isTS = isFileTS(filePath) && hasUnstableTSConfigFlag;
+
+    const isBun = isRunningInBun();
+
+    const isDeno = isRunningInDeno();
+
+    if (isTS && !isDeno && !isBun) {
+
+        const createJiti = await import("jiti").then(jitiModule => jitiModule.default, () => {
+            throw new Error("The 'jiti' library is required for loading TypeScript configuration files. Make sure to install it.");
+        });
+
+        /*
+         * Disabling `moduleCache` allows us to reload a
+         * config file when the last modified timestamp changes.
+         */
+
+        const jiti = createJiti(__filename, { moduleCache: false });
+
+        const config = typeof jiti?.import === "function" ? await jiti.import(fileURL.href) : await jiti(fileURL.href);
```

Comment:
`jiti.import()` only exists in Jiti 1.21.x or later: this is what we are testing in this PR. `jiti()` is always defined but it is not covered by unit tests, so I tried manually installing older vesions of Jiti and see if the tests were still passing when using `jiti()` to import the config. It turned out that all tests would still pass with `jiti v1.18` or later but some of them would fail in older versions, with more and more tests failing when an even older version is installed.

Since the `jiti()` call is currently untested and it may or may not work with older version of Jiti, I'd like to suggest that we only use `jiti.import` and that we throw an error instead of defaulting to calling `jiti()` when `jiti.import` is not a function. In the error message, we could for example tell users to install the latest version of `jiti` (currently v1). As noted in https://github.com/eslint/eslint/pull/18134#discussion_r1692629579, this should be also forward-compatible with Jiti v2.